code-challenges.test.js

```diff
@@ -14,33 +17,148 @@
+//     expect(remainderList(hodgepodge2)).toEqual([ 2, 1, -1 ])
+//  })
+// })
+
+//RED - Good failure
+// FAIL  ./code-challenges.test.js
+// remainderList
+//   ✕ creates a function that takes a mixed data array and returns an array of only the remainders of the number divided by 3 (1 ms)
+
+// ● remainderList › creates a function that takes a mixed data array and returns an array of only the remainders of the number divided by 3
+
+//   ReferenceError: remainderList is not defined
+
+
 // b) Create the function that makes the test pass.
+//PseudoCode
+//Input: const hodgepodge1 = [23, "Heyyyy!", 45, -10, 0, "Yo", false]
+// const hodgepodge2 = [5, "Hola", 43, -34, "greetings", true]
+//Output: [ 2, 0, -1, 0 ], [ 2, 1, -1 ]
+//Process: Create a function aclled remainderList and pass in array as a parameter
+//use the .filter method to filter through the array to find the data type of number, specify data type for the target data type using 'typeof', specify number
+//map through the array to find the data type from above and divide by 3
+//output is undefined
+
+//const hodgepodge1 = [23, "Heyyyy!", 45, -10, 0, "Yo", false]
+// const hodgepodge2 = [5, "Hola", 43, -34, "greetings", true]
+
+// const remainderList = (array) => {
+//   array.filter(item => typeof item === "number").map(number => number % 3)
+// }
+// console.log(remainderList(hodgepodge1))
+
+
+
 
 // --------------------3) Create a function that takes in an array of numbers and returns the sum of all the numbers cubed.
 
 // a) Create a test with an expect statement using the variables provided.
 
-const cubeAndSum1 = [2, 3, 4]
+// const cubeAndSum1 = [2, 3, 4]
 // Expected output: 99
-const cubeAndSum2 = [0, 5, 10]
+// const cubeAndSum2 = [0, 5, 10]
 // Expected output: 1125
 
+
+// describe("cubedList", () => {
+//     it("creates a function that takes an array of numbers and returns the sum of all the  numbers cubed", () => {
+//     expect(cubedList(cubeAndSum1)).toEqual(99)
+//     expect(cubedList(cubeAndSum1)).toEqual(1125)
+//   })
+// })
+
+
+//RED- Good failure
+// FAIL  ./code-challenges.test.js
+//   cubedList
+//     ✕ creates a function that takes an array of numbers and returns the sum of all the  numbers cubed (1 ms)
+
+//   ● cubedList › creates a function that takes an array of numbers and returns the sum of all the  numbers cubed
+
+//     ReferenceError: cubedList is not defined
+
+
+
+
+
 // b) Create the function that makes the test pass.
+//PseudoCode
+//input: // const cubeAndSum1 = [2, 3, 4]
+// const cubeAndSum2 = [0, 5, 10]
+//output 99, 1125
+//process
+//create a function called cubedList that takes in an array as a parameter
+//create a variable that gives the cubed number a place to go
+//iterate through the array using a for loop
+//call on newNum to equal the array at each index to be multiplied by 3
+
+// const cubedList => (array) => {
+//   let newNum = 0
+//   for(let i=0; array.length < i; i++) {
+//     newNum = array[i] * 3;
```

Comment:
Very close. Remember you want the number to the 3rd power, so it would be array[i] ** 3. Also, you want to continuously update newNum, so you would have to do `newNum = newNum + array[i] ** 3`